07_distributed_systems/01_intro/client/client.go

```diff
@@ -0,0 +1,129 @@
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io"
+	"kv_store/protocol"
+	"net/http"
+	"os"
+	"regexp"
+	"time"
+)
+
+const (
+	SERVER = "127.0.0.1"
+	PORT   = "8000"
+	URL    = "http://" + SERVER + ":" + PORT
+)
+
+func panicOnError(err error) {
+	if err != nil {
+		panic(err)
+	}
+}
+
+// abozhenko for oz: Is it ok to use http at all?
```

Comment:
@alexbozhenko sure, why not? Of course, it couples you to TCP, and you may not actually need much HTTP functionality (ie request line, HTTP verbs, status codes, headers etc) but otherwise it's fine. gRPC uses HTTP/2... not necessarily a perfect fit but avoids creating a new application layer protocol for the purpose, and theoretically allows more interoperability.